### PR TITLE
refactored deseq2 scripts

### LINKED ISSUE #698: FR: add explanation to rna-seq clustering
**Is your feature request related to a problem? Please describe.**
image

```
log_counts <- varianceStabilizingTransformation(dds, blind = TRUE)
sampleDistMatrix <- as.matrix(dist(t(assay(log_counts))))
```

> dds contains the counts, which are then transformed.
> Next, a distance matrix is computed with dist https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/dist

Explanation in layman's terms should be added :point_up:

### LINKED ISSUE #699: FR: more RNA-seq clusterings
Pearsson, Spearman. Good time to review the clustering code as well.

### PR DESCRIPTION
**Pull request by siebrenf**
**What problem is the PR solving / What's new?**
fixes #698 
fixes #699
fixed undocumented issue where deseq2 made each contrast for each assembly

**What did change**
- DESeq2 scripts moved into their own subdirectory :  `seq2science/scripts/deseq`
- repetitive/separate code chunks moved to `seq2science/scripts/deseq/utils.R`
- cluster script generates sample distance, and Pearson- & Spearman correlation heatmaps.
- cluster plots annotated in the multiQC.

**Checklist**
- [x] I made a PR to develop (not master)
- [x] If applicable: I updated the docs
- [x] I updated the CHANGELOG
- [ ] These changes are covered by the tests

### CODE REVIEW

File: seq2science/schemas/multiqc_config.yaml [hunk 1]
```diff
@@ -49,14 +49,20 @@ report_section_order:
...
+  Pearson_correlation_clustering:
+    order: -894
   deeptools_spearman:
     order: -896
   deeptools_pearson:
     order: -897
   deeptools_macs2_heatmap:
     order: -898
-  deeptools_macs2_heatmap:
+  deeptools_genrich_heatmap:
```
Comment (Maarten-vd-Sande) [inline seq2science/schemas/multiqc_config.yaml:65]: :smile:

File: seq2science/schemas/multiqc_config.yaml [hunk 2]
```diff
@@ -49,14 +49,20 @@ report_section_order:
...
     order: 1000
   dupRadar:
-    order: -895
+    order: -891
+  Sample_distance_clustering:
+    order: -892
+  Spearman_correlation_clustering:
+    order: -893
+  Pearson_correlation_clustering:
+    order: -894
```
Comment (Maarten-vd-Sande) [inline seq2science/schemas/multiqc_config.yaml:58]: It's minor, but this naming is sorta confusing with the deeptools clustering. Deeptools outputs this image:

image

Which is kinda the same, but based on reads over all regions. 

Should we call them `deseq_spearmap_correlation_clustering` and `deeptools_spearmap_correlation_clustering`?

Also I think the ordering should be that deseq2 comes after deeptools, since the deeptools plot is based on sliding window (no peak calling) and the deseq2 differential peaks (for the genomic workflows).

File: seq2science/scripts/deseq2/run_as_rule.R [hunk 3]
```diff
@@ -1,5 +1,3 @@
-#!/usr/bin/env Rscript
-
```
Comment (Maarten-vd-Sande) [inline seq2science/scripts/deseq2/run_as_rule.R:2]: why did you remove this?

**Review by Maarten-vd-Sande — APPROVED** [review]
Nice

File: seq2science/scripts/deseq2/run_as_rule.R (hunk 3 above)
Comment (siebrenf) [reply seq2science/scripts/deseq2/run_as_rule.R:2]: R's `source()` functions reads the sourced file inline. All objects and libraries are globals. It also means we only need to specify the shebang on the script(s) that are supposed to be called from the command line (pretty sure, I'll test this before merging). These files are supposed to be called from deseq.R only, never alone.

File: seq2science/schemas/multiqc_config.yaml (hunk 2 above)
Comment (siebrenf) [reply seq2science/schemas/multiqc_config.yaml]: Agreed on both counts. I've did some renaming in the qc.smk and the multiqc schema. Let me know if it was too much!

**Comment by siebrenf** [comment]
Original plot:
GRCh38 p13-DESeq2_sample_distance_clustering_mqc

New plots:
GRCh38 p13-DESeq2_pearson_correlation_clustering_mqc
GRCh38 p13-DESeq2_spearman_correlation_clustering_mqc

**Comment by siebrenf** [comment]
MultiQC order works, and DESeq heatmaps will use descriptive names if available (similar to the deepTools plots).
image

**Comment by siebrenf** [comment]
alright, it's ready. If you have any opinions on the MultiQC order or naming @Maarten-vd-Sande, now is the time!

**Comment by Maarten-vd-Sande** [comment]
Can you send me a multiqc report?

**Review by Maarten-vd-Sande — APPROVED** [review]
Cool :sunglasses:
